specutils/analysis/resample.py

```diff
@@ -0,0 +1,219 @@
+        # bin; negatives clipped
+        dl = (l_sup - l_inf).clip(0)
+
+        # This will only happen at the edges of lorig.
+        # Discard resampled bin if it's not fully covered (> 99%) by the
+        #  original bin -- only happens at the edges of the original bins
+        if 0 < dl.sum() < 0.99 * delta_fin:
+            dl = 0 * orig_lamb
+
+        resamp_mat[i, :] = dl
+
+    resamp_mat = resamp_mat / delta_fin
+
+    return resamp_mat
+
+
+def _nonuniform_matrix(orig_lamb, fin_lamb, extrapolate=False):
+    """
+    Compute re-sampling matrix R_o2r, useful to convert a spectrum sampled at
+    wavelengths lorig to a new grid lresamp. Here, there is no necessity to
+    have constant gris as on :func:`_uniform_matrix`. This is adapted from code
+    created by the SEAGal Group.
+
+    .. warning:: lorig and lresam MUST be on ascending order!
+
+    Parameters
+    ----------
+    orig_lamb : array_like
+        Original spectrum lambda array.
+    fin_lamb : array_like
+        Spectrum lambda array in which the spectrum should be sampled.
+    extrapolate : boolean, optional
+        Extrapolate values, i.e. values for fin_lamb < orig_lamb[0] are set to
+        match orig_lamb[0] and values for fin_lamb > orig_lamb[-1] are set to
+        match orig_lamb[-1].
+
+    Returns
+    -------
+    resample_mat : ndarray
+        Resample matrix.
+    """
+    matrix = np.zeros(shape=(len(fin_lamb), len(orig_lamb)))
+
+    # Define lambda ranges (low, upp) for original and resampled.
+    lo_low = np.zeros(len(orig_lamb))
+    lo_low[1:] = (orig_lamb[1:] + orig_lamb[:-1]) / 2
+    lo_low[0] = orig_lamb[0] - (orig_lamb[1] - orig_lamb[0]) / 2
+
+    lo_upp = np.zeros(len(orig_lamb))
+    lo_upp[:-1] = lo_low[1:]
+    lo_upp[-1] = orig_lamb[-1] + (orig_lamb[-1] - orig_lamb[-2]) / 2
+
+    lr_low = np.zeros(len(fin_lamb))
+    lr_low[1:] = (fin_lamb[1:] + fin_lamb[:-1]) / 2
+    lr_low[0] = fin_lamb[0] - (fin_lamb[1] - fin_lamb[0]) / 2
+
+    lr_upp = np.zeros(len(fin_lamb))
+    lr_upp[:-1] = lr_low[1:]
+    lr_upp[-1] = fin_lamb[-1] + (fin_lamb[-1] - fin_lamb[-2]) / 2
+
+    # Iterate over resampled fin_lamb vector
+    for i in range(len(fin_lamb)):
+
+        # Find in which bins fin_lamb bin within orig_lamb bin
+        bins_resamp = np.where((lr_low[i] < lo_upp) & (lr_upp[i] > lo_low))[0]
+
+        # On these bins, eval fraction of resampled bin is within original bin.
+        for j in bins_resamp:
+            aux = 0
+
+            d_lr = lr_upp[i] - lr_low[i]
+            d_lo = lo_upp[j] - lo_low[j]
+            d_ir = lo_upp[j] - lr_low[i]  # common section on the right
+            d_il = lr_upp[i] - lo_low[j]  # common section on the left
+
+            # Case 1: resampling window is smaller than or equal to the
+            # original window. This is where the bug was: if an original bin
+            # is all inside the resampled bin, then all flux should go into it,
```

Comment:
Is this a comment ported from the old code?

Reply:
Yes, removing.